core/src/test/java/com/yahoo/oak/MapApiTest.java

```diff
@@ -31,7 +31,7 @@ public void init() {
         int maxItemsPerChunk = 2048;
         OakMapBuilder<Integer, Integer> builder = OakCommonBuildersFactory.getDefaultIntBuilder()
                 .setChunkMaxItems(maxItemsPerChunk);
-        oak = builder.build();
+        oak = builder.buildOrderedMap();
```

Comment:
This is actually quite a critical test to add for OakHashMap execution as well. This is after all MapApiTest and API is what is done here...

I understand that this is absolutely not trivial task, due to subMap and iterators being used. However, I would encourage some creative thinking (only for this test). Can we check the instance for being OakMap or OakHashMap and act accordingly?

OakHashMap supports iterators (via keySet/entrySet/values) just only the full iterators and not a range iterators. So this code path better to be executed in some test. Is it possible to add something here?

Reply:
I strongly agree.
I suggest splitting this test into two. One for the common API (which will be parametrized) and another one just for the unique OakMap API.